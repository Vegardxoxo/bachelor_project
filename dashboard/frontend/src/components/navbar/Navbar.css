img {
    width: 350px;
    height: 65px;
}

#navBar {
    background-color: white;
    width: 100%;
    position: fixed;
    top: 0;
    left: 0;
    padding-top: 1%;
    padding-bottom: 1%;
    text-align: center;
    display: flex;
    justify-content: center;
    z-index: 9;
}

#menuIcon {
    padding-left: 620px;
    color: #313233;

}

.nav-menu {
    background-color: #ffffff;
    width: 400px;
    height: 100vh;
    display: flex;
    justify-content: center;
    position: fixed;
    top: 0;
    right: -100%;
    transition: 850ms;
}

.nav-menu.active {
    right: 0;
    transition: 350ms;
}

.nav-text {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    padding: 8px 0px 8px 16px;
    list-style: none;
    height: 60px;
}

.nav-text a {
    text-decoration: none;
    color: black;
    font-family: Source Sans Pro, sans-serif;
    font-size: 25px;
    width: 95%;
    height: 100%;
    display: flex;
    align-items: center;
    padding: 0 16px;
    border-radius: 4px;
}

.nav-text a:hover {
    font-weight: bold;
    color: #005aa7
}

.nav-menu-items {
    width: 100%;
}

.navbar-toggle {
    background-color: white;
    width: 100%;
    height: 80px;
    display: flex;
    justify-content: flex-start;
    align-items: center;
}

#spanning {
    margin-left: -22px;
}
