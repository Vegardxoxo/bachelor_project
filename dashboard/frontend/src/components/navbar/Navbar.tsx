import React, {useState} from 'react';
import Grid from '@mui/material/Grid';
import MenuIcon from '@mui/icons-material/Menu';
import logo from '../../assets/TrondheimKommuneLogo.png';
import './Navbar.css';
import {NavLink} from 'react-router-dom';
import ClearIcon from '@mui/icons-material/Clear';
import {SidebarData} from './Sidebar';


function Navbar() {
    const [sidebar, setSidebar] = useState(false);
    const showSidebar = () => setSidebar(!sidebar);

    /*Renders the navbar. The sidebar will be displayed when the sidebar state is true, and hidden when it's false.*/
    return (
        <nav data-testid="navbar">
            <Grid id="navBar">
                <Grid id="logoTrondheim">
                    <NavLink to="/" className="logo">
                        <img src={logo} alt="logo"/>
                    </NavLink>
                </Grid>
                <Grid id="menuIcon">
                    <NavLink to="#" className="menu-bars">
                        <MenuIcon
                            data-testid="menuIcon"
                            sx={{fontSize: 50}}
                            onClick={showSidebar}
                            style={{color: '#302d2d'}}
                        />
                    </NavLink>
                </Grid>
                <nav
                    data-testid="sidebar"
                    className={sidebar ? 'nav-menu active' : 'nav-menu'}>
                    <ul className="nav-menu-items" onClick={showSidebar}>
                        <li className="navbar-toggle">
                            <NavLink to="#" className="menu-bars">
                                <ClearIcon
                                    data-testid="clearIcon"
                                    sx={{fontSize: 40}}
                                    style={{color: '#302d2d'}}
                                />
                            </NavLink>
                        </li>

                        {SidebarData.map((item, index) => {
                            return (
                                <li key={index} className={item.cName}>
                                    <NavLink to={item.path} onClick={item.onClick}>
                                        <span id="spanning">{item.title}</span>
                                    </NavLink>
                                </li>
                            );
                        })}
                    </ul>
                </nav>
            </Grid>
        </nav>
    );
}

export default Navbar;
