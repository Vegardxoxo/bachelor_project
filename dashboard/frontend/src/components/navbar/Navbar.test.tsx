import {cleanup, render, screen} from '@testing-library/react';
import Navbar from './Navbar';
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import {MemoryRouter} from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import renderer from "react-test-renderer";

describe('Testing render, Navbar', () => {
    beforeEach(() => {
        render(
            <MemoryRouter>
                <Navbar/>
            </MemoryRouter>
        );
    });

    afterEach(() => {
        cleanup();
    });

    it('renders navbar without crashing', () => {
        expect(screen.getByTestId('navbar')).toBeInTheDocument();
    });

    it('matches snapshot', async () => {
        const testRenderer = renderer.create(
            <MemoryRouter>
                <Navbar/>
            </MemoryRouter>
        );
        expect(testRenderer.toJSON()).toMatchSnapshot();
    })

    it('renders sidebar when menu icon is clicked', () => {
        const menuIcon = screen.getByTestId('menuIcon');
        userEvent.click(menuIcon);
        const navMenu = screen.getByTestId('sidebar');
        expect(navMenu).toHaveClass('nav-menu active');
    });

    it('hides sidebar when clear icon is clicked', () => {
        const menuIcon = screen.getByTestId('menuIcon');
        userEvent.click(menuIcon);
        const clearIcon = screen.getByTestId('clearIcon');
        userEvent.click(clearIcon);
        const navMenu = screen.getByTestId('sidebar');
        expect(navMenu).toHaveClass('nav-menu');
    });

    it('renders sidebar links correctly', () => {
        const menuIcon = screen.getByTestId('menuIcon');
        userEvent.click(menuIcon);
        const sidebarLinks = screen.getAllByRole('link', {hidden: true});
        expect(sidebarLinks.length).toBeGreaterThan(0);
    });
});
