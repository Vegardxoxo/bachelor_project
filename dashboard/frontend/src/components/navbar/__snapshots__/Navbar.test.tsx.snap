// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing render, Navbar matches snapshot 1`] = `
<nav
  data-testid="navbar"
>
  <div
    className="MuiGrid-root css-vj1n65-MuiGrid-root"
    id="navBar"
  >
    <div
      className="MuiGrid-root css-vj1n65-MuiGrid-root"
      id="logoTrondheim"
    >
      <a
        aria-current="page"
        className="logo active"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="logo"
          src="test-file-stub"
        />
      </a>
    </div>
    <div
      className="MuiGrid-root css-vj1n65-MuiGrid-root"
      id="menuIcon"
    >
      <a
        aria-current="page"
        className="menu-bars active"
        href="/"
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-dqscbn-MuiSvgIcon-root"
          data-testid="menuIcon"
          focusable="false"
          onClick={[Function]}
          style={
            Object {
              "color": "#302d2d",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </a>
    </div>
    <nav
      className="nav-menu"
      data-testid="sidebar"
    >
      <ul
        className="nav-menu-items"
        onClick={[Function]}
      >
        <li
          className="navbar-toggle"
        >
          <a
            aria-current="page"
            className="menu-bars active"
            href="/"
            onClick={[Function]}
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1ha5wk2-MuiSvgIcon-root"
              data-testid="clearIcon"
              focusable="false"
              style={
                Object {
                  "color": "#302d2d",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </a>
        </li>
        <li
          className="nav-text"
        >
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
          >
            <span
              id="spanning"
            >
              Mitt dashboard
            </span>
          </a>
        </li>
        <li
          className="nav-text"
        >
          <a
            className=""
            href="/lisensportal"
            onClick={[Function]}
          >
            <span
              id="spanning"
            >
              Lisensportalen
            </span>
          </a>
        </li>
        <li
          className="nav-text"
        >
          <a
            className=""
            href="/leaderboard"
            onClick={[Function]}
          >
            <span
              id="spanning"
            >
              Ledertavle
            </span>
          </a>
        </li>
        <li
          className="nav-text"
        >
          <a
            className=""
            href="/minside"
            onClick={[Function]}
          >
            <span
              id="spanning"
            >
              Min side
            </span>
          </a>
        </li>
        <li
          className="nav-text"
        >
          <a
            className=""
            href="/FAQ"
            onClick={[Function]}
          >
            <span
              id="spanning"
            >
              Ofte stilte spørsmål
            </span>
          </a>
        </li>
        <li
          className="nav-text"
        >
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
          >
            <span
              id="spanning"
            >
              Logg ut
            </span>
          </a>
        </li>
      </ul>
    </nav>
  </div>
</nav>
`;
